docs-src/guide-rpc-connect.scrbl

```diff
@@ -0,0 +1,519 @@
+@subsection[#:tag "rpc-connect-js-lib-walkthrough-rpc"]{The @bold{rpc} function}
+
+Now we'll construct the workhorse of our client: @jsin{rpc}.
+
+@js{
+  const rpc = async (m, ...args) => {
+    const lab = `RPC ${m} ${JSON.stringify(args)}`
+    console.log(`${lab}`);
+    const ans = await call(m, args);
+    console.log(`${lab} ==> ${JSON.stringify(ans)}`);
+    return ans;
+  };
+}
+
+This simple function logs the request it's about to make, makes it, then logs
+and returns the response.
+
+Note the function arguments: @jsin{m} represents the @tech{RPC} method (e.g.
+@urlin{/stdlib/balanceOf}), and @jsin{...args} makes use of JavaScript's
+support of
+@link["https://en.wikipedia.org/wiki/Variadic_function#In_JavaScript"]{variadic functions}
+to capture arguments which are then forwarded to the operation represented by
+@jsin{m}.
+
+If your language doesn't support variadics the solution is simple: just make
+the second argument a list-type!
+
+
+@;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+@subsection[
+  #:tag "rpc-connect-js-lib-walkthrough-rpc-callbacks"
+  ]{The @bold{rpcCallbacks} function}
+
+The @jsin{rpcCallbacks} function is somewhat more involved.
+
+@js{
+  const rpcCallbacks = async (m, arg, cbacks) => {
+    const vals = {};
+    const meths = {};
+    for (const k in cbacks) {
+      const v = cbacks[k];
+      if ( v instanceof Function ) {
+        meths[k] = true;
+      } else {
+        vals[k] = v;
+      }
+    }
+    return new Promise((resolve, reject) => (async () => {
+      let p = rpc(m, arg, vals, meths);
+      while (true) {
+        try {
+          const r = await p;
+          switch ( r.t ) {
+            case 'Done': {
+              return resolve(r.ans);
+            }
+            case 'Kont': {
+              const { kid, m, args } = r;
+              const ans = await cbacks[m](...args);
+              p = rpc(`/kont`, kid, ans);
+              break;
+            }
+            default:
+              throw new Error(`Illegal callback return: ${JSON.stringify(r)}`);
+          }
+        } catch (e) {
+          return reject(e);
+        }
+      }
+    })());
+  };
+}
+
+We begin by bisecting the @jsin{cbacks} argument into a hash of values and a
+hash of methods. This effectively splits static fields (@jsin{vals}) in a
+@tech{participant interact interface} from callable functions (@jsin{meths}).
+
+Then we spawn an asynchronous task in the form of a @jsin{Promise} which will
+advance the state of the @DApp by continuously invoking PII functions until
+the server responds with a message indicating completion.
+
+In the first @tech{RPC} request, @jsin{m} will be a call to a method such as
+@urlin{/backend/Alice} or @urlin{/backend/AnotherParticipant} which begins the
+cycle.
+
+Subsequent requests are directed to the @urlin{/kont} method (which is another
+name for continuation). This is the only "semi-magical" ingredient in our
```

Comment:
I don't like using the term "magic" in connection with computer systems. I'd say something more precise like... this is the only part of an RPC client that operates automatically without the frontend program's intervention. All of the other functions returned immediately, but backends block and generate asynchronous callbacks... just like when you use the Reach standard library directly.